### PR TITLE
[FLINK-35581] Remove comments from the code related to ingestDB

### PR DESCRIPTION
**Pull request by mayuehappy**
Revert commit 268a3080 "[FLINK-31238] Deactivate parts of the code until new FRocksDB release is available."

## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follow [the conventions for tests defined in our code quality guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#7-testing).

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by mayuehappy** [comment]
Now, we've updated the Frocksdb version to V8.10 , So I think we can remove the comments in the code to make ingestDB available. @rkhachatryan @StefanRRichter  , can you please take a look at this PR ?

**Comment by rkhachatryan** [comment]
@flinkbot run azure

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOperationUtils.java
Comment (StefanRRichter) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOperationUtils.java:136]: nit: please change the order of parameters (first the list, then ICloseableRegistry) so that it's consistent with the order of `createColumnFamily`.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java
Comment (StefanRRichter) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java:653]: @rkhachatryan this line looks like it's undoing a change that you introduced when porting ClosableRegistry changes. In the long term, we still want that version because the `RegisteredStateMetaInfoBase.Key` comes from a fix that I committed. So I wonder what's the best order and strategy? Should we stop doing this uncommenting and rather create a branch that first applies the remaining bugfixes and THEN uncomments, or do we start by uncommenting even though it potentially undoes changes like this that we need to redo again. I almost think the first approach is better, wdyt?

Comment (StefanRRichter) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java:653]: Discussed offline with @rkhachatryan and we are ok with proceeding with this PR first and then we'll just re-apply all required code for the bugfixes.

**Review by StefanRRichter — APPROVED** [review]
LGTM but the code after this commit has known bugs for which Roman / me will port the commits from an internal branch after this was merged.

**Review by rkhachatryan — APPROVED** [review]
Thanks for the PR, LGTM

**Comment by mayuehappy** [comment]
@rkhachatryan @StefanRRichter I've updated the PR , can you help merge it after the CI pass ?

**Comment by StefanRRichter** [comment]
@flinkbot run azure

**Pull request merged**
